etcdserver/api/membership/cluster.go

```diff
@@ -567,7 +591,7 @@ func (c *RaftCluster) SetVersion(ver *semver.Version, onSet func(*zap.Logger, *s
 	}
 	oldVer := c.version
 	c.version = ver
-	mustDetectDowngrade(c.lg, c.version)
+
```

Comment:
why this is removed?

Reply:
`SetVersion` is called when restarting server. Each history cluster version wal log will be applied. Here is a case. If the cluster upgrades first then downgrades, this check will fail when applying old version wal log. For example, the version change is 3.4->3.5->3.4. After finishing downgrade, restarting server will fail applying second 3.4 wal log.

Reply:
Discussed offline, it is difficult to argue and guarantee correctness when there are both v2 and v3 raft entries for cluster info (downgradeInfo and cluster version). We should migrate cluster related stuff to use v3 backend. Just opened #11380.

Reply:
Theoretically, replaying raft log entries from a previous state should always result in the same result. I believe we are violating the rules by directly reading cluster info from v3backend / v2store during cluster recovery.